src/widgets/ProfileWidget/ProfileWidget.tsx

```diff
@@ -1,39 +1,103 @@
-  const ensAddress = useEnsAddress(account || '');
+interface ProfileWidgetProps {
+  className?: string;
+}
+
+const ProfileWidget: FC<ProfileWidgetProps> = ({ className = '' }) => {
+  const { collectionToken, collectionImage } = useAppSelector((state) => state.config);
   const { avatarUrl } = useAppSelector((state) => state.user);
+  const { tokens } = useAppSelector((state) => state.balances);
+  const { isLoading, ownedTokenMeta } = useAppSelector((state) => state.profile);
+  const { currencyTokenInfo } = useAppSelector((state) => state.metadata);
+
+  const [ownedNfts, setOwnedNfts] = useState<CollectionTokenInfo[]>([]);
+  const [searchValue, setSearchValue] = useState('');
+
+  const { library, account, deactivate } = useWeb3React<Web3Provider>();
+  const { account: accountId } = useParams();
+
+  const dispatch = useAppDispatch();
+  const ensAddress = useEnsAddress(account || '');
+
+  const filteredNfts = useMemo(() => ownedNfts.filter(nft => filterNftBySearchValue(searchValue, nft)), [ownedNfts, searchValue]);
+
+  useEffect(() => {
+    // TODO: support loading the tokens of other users
+    if (tokens && library) {
+      dispatch(fetchOwnedTokenMeta({ library, collectionToken, tokenIds: tokens }));
+    }
+  }, [library, collectionToken, tokens]);
+
+  useEffect(() => {
+    if (ownedTokenMeta) {
+      setOwnedNfts(ownedTokenMeta);
+    }
+  }, [ownedTokenMeta, isLoading]);
 
   const handleDisconnectClick = () => {
     deactivate();
   };
 
   return (
-    <div className="profile-widget">
+    <div className={`profile-widget ${className}`}>
       <ProfileHeader
-        backgroundImage={collectionImage}
         avatarUrl={avatarUrl}
+        backgroundImage={collectionImage}
         ensAddress={ensAddress}
         address={account || ''}
+        showLogOutButton={account === accountId}
         onLogoutButtonClick={handleDisconnectClick}
       />
       <div className="profile-widget__button-group-container">
         <div className="profile-widget__button-group">
           <Button text="NFTs" className="profile-widget__button-group__button profile-widget__button-group__button--is-active" />
-          <Button text="ACTIVITY" className="profile-widget__button-group__button" />
-          <Button text="LISTED" className="profile-widget__button-group__button" />
+          <Button text="Activity" className="profile-widget__button-group__button profile-widget__button-group__button--is-uppercase" disabled />
+          <Button text="Listed" className="profile-widget__button-group__button profile-widget__button-group__button--is-uppercase" disabled />
+        </div>
+      </div>
+      <div className="profile-widget__content">
+        <div className="profile-widget__search-bar-container">
+          <IconSearch className="profile-widget__search-bar-icon" />
+          <Input className="profile-widget__search-bar" placeholder="Search NFT" onChange={e => setSearchValue(e.target.value)} />
+        </div>
```

Comment:
```suggestion
        <SearchInput
          placeholder="Search NFT"
          onChange={(e) => setSearchInput(e.target.value)}
          value={searchInput}
          className="profile-widget__search-input"
        />
```

With styling:

```
&__search-input {
        max-width: 23.5rem;

        @include for-size(phone-only) {
            max-width: 100%;
        }
    }
```

Reply:
Do you know what.. thanks for this comment here.. I'd only seen the design from the issue but this made me think I needed to double check the Figma..
I'd not seen the desktop design.. My bad.

Reply:
> When you go to another account's profile page it still loads your own nft's. Are you going to address this or do in another ticket?

I would say tackle with indexer integration?